plugins/services/src/js/structs/Framework.js

```diff
@@ -20,6 +20,14 @@ module.exports = class Framework extends Application {
     return this.get("slave_ids");
   }
 
+  getTasks() {
```

Comment:
I hear you and you have a good argument. However, consistency should trump most arguments, especially when the tradeoffs are low. Our codebase has become rather inconsistent and hence, complex and that adds significant overhead to onboarding and reviewing. I agree we should change the legacy, but let's plan that change effectively.